test/FIFOMarket.test.ts

```diff
@@ -28,48 +58,346 @@ const setupTestLocal = async (
   return {
     hre,
     contracts,
+    removal,
+    fifoMarket,
     ...rest,
+    listedRemovalIds: tokenIds,
   };
 };
 
 describe('FIFOMarket', () => {
+  describe('initialization', () => {
+    describe('roles', () => {
+      (
+        [{ role: 'DEFAULT_ADMIN_ROLE' }, { role: 'ALLOWLIST_ROLE' }] as const
+      ).forEach(({ role }) => {
+        it(`will assign the role ${role} to the deployer and set the DEFAULT_ADMIN_ROLE as the role admin`, async () => {
+          const { fifoMarket, hre } = await setupTest();
+          expect(
+            await fifoMarket.hasRole(
+              await fifoMarket[role](),
+              hre.namedAccounts.admin
+            )
+          ).to.be.true;
+          expect(await fifoMarket.getRoleAdmin(await fifoMarket[role]())).to.eq(
+            await fifoMarket.DEFAULT_ADMIN_ROLE()
+          );
+          expect(
+            await fifoMarket.getRoleMemberCount(await fifoMarket[role]())
+          ).to.eq(1);
+        });
+      });
+    });
+  });
+  describe('role access', () => {
+    describe('roles', () => {
+      describe('DEFAULT_ADMIN_ROLE', () => {
```

Comment:
since this is printed in the `foreach` below, should we remove this describe? That way tou can use multiple roles in the array below

Reply:
if you don't want to `forEach` the roles, we should probably remove it from the object used below

Reply:
ok, I pulled out this `forEach` and the `roles` describe block to simplify this.  we can add a loop back in if we find ourselves adding additional tests for a given role